lib/request.js

```diff
@@ -1,18 +1,28 @@
 const http = require('http');
 const https = require('https');
+const zlib = require('zlib');
 
 module.exports = (options) => {
     options = Object.assign({
         protocol: 'https:'
     }, options);
     const httpModule = options.protocol === 'https:' ? https : http;
     return new Promise((resolve, reject) => {
-        const chunks = [];
         const request = httpModule.request(options, (response) => {
-            response.on('data', (chunk) => {
+            let bodyStream;
+            const chunks = [];
+            const encoding = response.headers && response.headers['content-encoding'];
+            if (encoding === 'gzip' || encoding === 'deflate') {
+                response.on('error', reject);
+                bodyStream = response.pipe(zlib.createUnzip());
+            } else {
+                bodyStream = response;
+            }
+            bodyStream.on('error', reject);
```

Comment:
Good point. I think Node.js takes care of removing event handlers from a stream with error. Then `chunks` array should also be GCed because it's not referenced anymore.

But [I did some experiment](https://gist.github.com/shuhei/06d585e40fe1d1536c1d20522c160c61) because I was not 100% sure. I couldn't find any memory leaks here.